app/src/main/java/org/simple/clinic/teleconsultlog/prescription/patientinfo/TeleconsultPatientInfoEffectHandler.kt

```diff
@@ -16,9 +19,18 @@ class TeleconsultPatientInfoEffectHandler @Inject constructor(
     return RxMobius
         .subtypeEffectHandler<TeleconsultPatientInfoEffect, TeleconsultPatientInfoEvent>()
         .addTransformer(LoadPatientProfile::class.java, loadPatientProfile())
+        .addTransformer(LoadPrescriptionDate::class.java, loadPrescriptionDate())
         .build()
   }
 
+  private fun loadPrescriptionDate(): ObservableTransformer<LoadPrescriptionDate, TeleconsultPatientInfoEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .map { LocalDate.now(userClock) }
```

Comment:
While this works, I prefer to add the current timestamp in an event when it is generated by the user (if it's a UI event) and then derive the date from it in the effect handler when it needs to be used instead of adding a separate effect for loading the current date.

But this is a good approach if the source trigger for this effect is not a UI event.

Reply:
Removed this after placing the `prescriptionDate` as factory parameter for the model.